### PR TITLE
Cache charmcraft pack container, skip unstable tests except on schedule

### PR DESCRIPTION
**Pull request by carlcsaposs-canonical**
Ported from https://github.com/canonical/mysql-k8s-operator/pull/146

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -13,87 +16,58 @@ jobs:
...
 
-  integration-client-relations:
-    name: Integration tests for client relations
+  integration-test:
+    strategy:
+      fail-fast: false
+      matrix:
+        tox-environments:
+          - backend-integration
+#          - client-relation-integration
```
Comment (marceloneppel) [inline .github/workflows/ci.yaml:35]: Did you comment this environment because it is unstable?

**Review by marceloneppel — APPROVED** [review]

File: tox.ini [hunk 2]
```diff
@@ -142,10 +124,21 @@ deps =
...
-    -r{toxinidir}/requirements.txt
+    -r {tox_root}/requirements.txt
+commands =
+    pytest -v --tb native --log-cli-level=INFO -s --durations=0 {posargs} {[vars]tests_path}/integration/bundle/test_bundle.py
+
+[testenv:read-distribution-integration]
+description = Run integration tests for legacy relations
+deps =
+    pytest
+    juju==2.9.11 # juju 3.0.0 has issues with retrieving action results
```
Comment (taurus-forever) [inline tox.ini:135]: we should consider switch to 2.9.38.1

**Review by taurus-forever — APPROVED** [review]

**Comment by dragomirp** [comment]
> @marceloneppel Are the CI test failures expected?

Looks to me that we are pulling an old revision for the `tls-certificates-operator` and setting its series to focal. Looking at its [charmhub page](https://charmhub.io/tls-certificates-operator), it has a stable channel now and has switched to jammy.

**Comment by marceloneppel** [comment]
> @marceloneppel Are the CI test failures expected?

Thanks to Dragomir finding the errors were fixed. I'm still checking the instability in the bundle test job.

**Review by dragomirp — APPROVED** [review]

**Pull request merged**